more_itertools/more.py

```diff
@@ -1441,17 +1442,11 @@ def divide(n, iterable):
     """
     if n < 1:
         raise ValueError('n must be at least 1')
-
-    seq = tuple(iterable)
+    seq = iterable if hasattr(iterable, '__lengh__') else tuple(iterable)
```

Comment:
This won't do what we want for things like `dict`s. They have `__length__`, but won't allow for slicing.

Really what we want to check for is whether the iterable supports slicing. Checking whether something is an instance of `collections.abc.Sequence` could work here, or trying to retrieve a `slice(0)` and catching an exception.